teuthology/orchestra/remote.py

```diff
@@ -276,11 +277,20 @@ def write_file(self, path, data, sudo=False, mode=None, owner=None,
         :param owner:   set file owner if provided
         :param mkdir:   preliminary create the file directory, defaults False
         :param append:  append data to the file, defaults False
+        :param bs:      write up to N bytes at a time, defaults 512
+        :param offset:  number of bs blocks to seek to in file, defaults 0
+        :param sync:    sync file after write is complete if provided
         """
         dd = 'sudo dd' if sudo else 'dd'
         args = dd + ' of=' + path
         if append:
             args += ' conv=notrunc oflag=append'
+        if bs:
+            args += ' bs=' + str(bs)
+        if offset:
+            args += ' seek=' + str(offset)
+        if sync:
+            args += ' conv=sync'
```

Comment:
I don't remember exactly if the `conv` can be used multiple times in dd, if not we must take into consideration `notrunct` when `append` needed. Btw, one of use case for a unit test.